runtime/tests/checker/entitlements_test.go

```diff
@@ -7312,3 +7315,57 @@ func TestCheckEntitlementOptionalChaining(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestCheckEntitlementMissingInMap(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("missing type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+        access(all) entitlement X
+        access(all) entitlement mapping M {
+            X -> X
+            NonExistingEntitlement -> X
+        }
+        access(all) struct S {
+            access(M) var foo: auth(M) &Int
+            init() {
+                self.foo = &3 as auth(X) &Int
+                var selfRef = &self as auth(X) &S;
```

Comment:
Here and below: maybe remove unnecessary semicolons.

Can the code here simplified to just what is needed, or is is already a minimal test?

Reply:
Yes, the use of the entitlement and the member access were necessary to reproduce the crash; the reported errors here were already reported, we just continued checking after that point and then crashed.